### PR TITLE
kubectl-klock: 0.5.0 -> 0.5.1

### PR DESCRIPTION
**Pull request by scm2342**
## Description of changes
* update to version 0.5.1
* enable completion via wrapper script

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kubectl-klock/default.nix [hunk 1]
```diff
@@ -1,17 +1,23 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGo121Module, fetchFromGitHub, makeWrapper }:
```
Comment (kirillrdy) [inline pkgs/applications/networking/cluster/kubectl-klock/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, makeWrapper }:
```

go_1_21 is default, also in future overrides need to be done in all-packages.nix as per https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults

File: pkgs/applications/networking/cluster/kubectl-klock/default.nix [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+  version = "0.5.1";
+
+  nativeBuildInputs = [ makeWrapper ];
 
   src = fetchFromGitHub {
     owner = "applejag";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-fR97rTMFwtqVH9wqKy1+EzKKg753c18v8VDCQ2Y69+s=";
+    sha256 = "sha256-q7Wq1mTjOB7tT66+LWIwwqltQcQSHD/VHLO7nCTZTQ0=";
```
Comment (kirillrdy) [inline pkgs/applications/networking/cluster/kubectl-klock/default.nix:13]:
```suggestion
    hash = "sha256-q7Wq1mTjOB7tT66+LWIwwqltQcQSHD/VHLO7nCTZTQ0=";
```

for SRI hashes `hash` is preferred

**Comment by kirillrdy** [comment]
@scm2342 can you adjust first commit, so that it only contains changes related to completion

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 268373` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubectl-klock</li>
  </ul>
</details>

lets wait for CI to pass

**Comment by kirillrdy** [comment]
@scm2342 thank you, sorry this PR got lost :-)

**Comment by scm2342** [comment]
@kirillrdy No worries at all - I should have requested review sooner. Thanks a lot for your work!

**Pull request merged**